### PR TITLE
Add support for Python 3.13

### PR DESCRIPTION
**Pull request by phofl**
Closes #xxxx

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by graingert** [comment]
> Basic Constraints of CA cert not marked critical

this is https://github.com/python/cpython/issues/107361  I'd recommend using https://trustme.readthedocs.io/en/latest/ to generate certs

**Review by jacobtomlinson — COMMENTED** [review]
🎉 thanks for handling this @phofl

File: distributed/security.py [hunk 1]
```diff
@@ -326,6 +326,8 @@ def _get_tls_context(self, tls, purpose):
             # IP addresses rather than hostnames
             ctx.check_hostname = False
 
+            ctx.verify_flags &= ~ssl.VERIFY_X509_STRICT
```
Comment (jrbourbeau) [inline distributed/security.py:329]: I've gone ahead and turned this new `VERIFY_X509_STRICT` flag off here (it was previously off by default in older versions of Python but was flipped on by default in Python 3.13). This seems not ideal, but also not that bad given that's what we've always done prior to Python 3.13. @jacobtomlinson (or anyone else) let me know if you have a better suggestion, I'm not verify familiar with our TLS security code paths.

File: continuous_integration/environment-3.13.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  - tblib
+  - toolz
+  - tornado
+  - zict  # overridden by git tip below
+  - zstandard
+  # Temporary fix for https://github.com/pypa/setuptools/issues/4496
+  - setuptools < 71
+  - pip:
+      - git+https://github.com/dask/dask
+      # - git+https://github.com/dask-contrib/dask-expr
```
Comment (jrbourbeau) [inline continuous_integration/environment-3.13.yaml:50]: I'm not sure why this is commented out. @phofl is this still needed?

```suggestion
      - git+https://github.com/dask/dask-expr
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phofl

File: continuous_integration/environment-3.13.yaml (hunk 2 above)
Comment (phofl) [reply continuous_integration/environment-3.13.yaml:50]: dask-expr pulls in pyarrow which wasn't compatible with python 3.13

File: distributed/security.py (hunk 1 above)
Comment (graingert) [reply distributed/security.py:329]: using https://github.com/python-trio/trustme to generate the certificates used for testing on the fly would get this to pass (as it sets the appropriate flags on the certificates and generally keeps these flags up to date)

of course, distributed users would also need to configure their certificates to meet these stronger security standards

File: continuous_integration/environment-3.13.yaml (hunk 2 above)
Comment (jrbourbeau) [reply continuous_integration/environment-3.13.yaml:50]: Ah, I see. `pyarrow=18` was recently released and has Python 3.13 support, so I think we should be okay to add this back in then

Comment (phofl) [reply continuous_integration/environment-3.13.yaml]: Yes definitely, I was waiting for that

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -174,6 +176,13 @@ jobs:
             || steps.cache.outputs.cache-hit != 'true'
           )
 
+      - name: Install PyArrow nightlies for Python 3.13
```
Comment (hendrikmakait) [inline .github/workflows/tests.yaml:179]: Do we still need this?

Comment (jrbourbeau) [reply .github/workflows/tests.yaml]: Nope, not anymore. Just removed it

**Review by jrbourbeau — APPROVED** [review]
Thanks @phofl

**Pull request merged**